### PR TITLE
Lipinski filter

### PR DESCRIPTION
**Pull request by j-adamczyk**
## Changes

Added Lipinski Rule of 5 filter. With `BaseFilter` class, it can serve as a base for issue https://github.com/scikit-fingerprints/scikit-fingerprints/issues/197.

## Checklist before requesting a review
- [x] Docstrings added/updated in public functions and classes
- [x] Tests added, reasonable test coverage (at least ~90%, `make test-coverage`)
- [x] Sphinx docs added/updated and render properly (`make docs` and see `docs/_build/index.html`)

### CODE REVIEW

File: skfp/preprocessing/filters/lipinski.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    Examples
+    --------
+    >>> from skfp.preprocessing import LipinskiFilter, MolToSmilesTransformer
+    >>> smiles = ["[C-]#N", "CC=O", "O=C(O)c1ccccc1c2ccc(cc2)Cn3c4cc(cc(c4nc3CCC)C)c5nc6ccccc6n5C"]
+    >>> filt = LipinskiFilter()
+    >>> filt
+    LipinskiFilter()
+
+    >>> filtered_mols = filt.transform(smiles)
+    >>> mol_to_smiles = MolToSmilesTransformer()
```
Comment (mjste) [inline skfp/preprocessing/filters/lipinski.py:64]: There's no need for MolToSmilesTransformer -  filter returns SMILES anyway.

**Review by mjste — CHANGES_REQUESTED** [review]
Add base_filter to docs

File: skfp/preprocessing/filters/lipinski.py (hunk 1 above)
Comment (j-adamczyk) [reply skfp/preprocessing/filters/lipinski.py:64]: Right, I forgot about that

**Review by mjste — CHANGES_REQUESTED** [review]
Add BaseFilter to docs

**Review by mjste — APPROVED** [review]

**Pull request merged**